lib/foreman_maintain/utils/service/abstract.rb

```diff
@@ -31,7 +31,11 @@ def inspect
 
       def matches?(service)
         if service.is_a? String
-          service == @name
+          if service.include?('@*')
```

Comment:
What if they only want to restart part of it? Shouldn't it just split on `@`?

Reply:
Define part of it? Or provide me an example of the case you are thinking of @ekohl ?

Reply:
I think something like `restart pulpcore-worker@1` to only restart the first one. (Probably makes more sense with dynflow).

Reply:
Well, the original code already supported restarting a specific worker, I went on a wild goose chase.

Reply:
Ah, interesting.
And I should not review stuff on my phone ;)

Reply:
However, I think with systemd you can pass the wildcard at any point after the `@` such as `dynflow-sidekiq@worker*` and should we support that?

I am questioning all this indirection and "service checking" against a known list of services rather than just passing things directly to systemd. I am asking myself, what is the benefit of checking the service passed to `service restart` against a known list of services?

Reply:
For restart, not much (but I personally also would never use FM for that).
For status, it's useful to only see a subset of services :woman_shrugging: 

But then I started reading https://www.freedesktop.org/software/systemd/man/latest/systemctl.html#Parameter%20Syntax and think your solution only goes half as far as it could (to be correct).

How about

```ruby
      def matches?(service)
        if service.is_a? String
          service == @name || File.fnmatch(service, @name)
        elsif service.instance_of?(self.class)
          service.name == @name
        else
          false
        end
      end
```

This now also allows `foreman-maintain service status --only dynflow-sidekiq@worker*` and friends.